### PR TITLE
Support empty component interfaces in `make_*` scripts. 

### PR DESCRIPTION
**Pull request by tribeiro**
Update `make_idl_file.py` and `make_salpy_lib.py` in order to support components with generic topics only.

### CODE REVIEW

**Comment by dmills-lsst** [comment]
Can you make the new code to add the dummy _Events.xml occur  only once , instead of duplicate

File: bin/make_idl_files.py [hunk 1]
```diff
@@ -41,6 +41,14 @@
     components = lsst.ts.xml.subsystems
 else:
     components = args.components
+    invalid_components = [
+        component for component in components if component not in lsst.ts.xml.subsystems
+    ]
+    if len(invalid_components):
```
Comment (r-owen) [inline bin/make_idl_files.py:47]: I found this a bit hard to read. I think it would be more idomatic to skip the len:
```
if invalid_components:
   ...
```
Another (more verbose) option is to explicitly compare the length against 0:
```
if len(invalid_components)> 0:
    ...
```

File: bin/make_salpy_libs.py [hunk 2]
```diff
@@ -19,5 +21,16 @@
...
+    component
+    for component in args.components
+    if component not in lsst.ts.xml.subsystems
+]
+if len(invalid_components):
+    raise RuntimeError(
+        f"The following components are not part of sal subsystems {invalid_components}. "
+        "Check spelling and try again."
+    )
+
```
Comment (r-owen) [inline bin/make_salpy_libs.py:34]: Same comments as above.

File: python/lsst/ts/sal/make_idl_file.py [hunk 3]
```diff
@@ -63,9 +63,22 @@ def copy_xml_files(self):
...
+            )
+            with open(dummy_xml_event_file, "w") as fp:
+                fp.write(
+                    """<?xml version="1.0" encoding="UTF-8"?>
+<?xml-stylesheet type="text/xsl" href="http://lsst-sal.tuc.noao.edu/schema/SALEventSet.xsl"?>
+<SALEventSet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://lsst-sal.tuc.noao.edu/schema/SALEventSet.xsd">
+</SALEventSet>
+"""
+                )
+
```
Comment (r-owen) [inline python/lsst/ts/sal/make_idl_file.py:81]: Please see if this `if not sal_xml_paths:` is needed -- especially the parts that writes the dummy event file -- and if not, just delete it. I think this worked around a bug in ts_sal's generator but I'm hoping that bug has been fixed.

It probably makes sense to print  the message, though I'd reword it a bit to make sure it sounds perfectly normal, for example "Note: SAL component {self.name} only uses generic topics".

File: python/lsst/ts/sal/make_salpy_lib.py [hunk 4]
```diff
@@ -80,9 +80,22 @@ def copy_xml_files(self):
...
+            )
+
+            with open(dummy_xml_event_file, "w") as fp:
+                fp.write(
+                    """<?xml version="1.0" encoding="UTF-8"?>
+<?xml-stylesheet type="text/xsl" href="http://lsst-sal.tuc.noao.edu/schema/SALEventSet.xsl"?>
+<SALEventSet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://lsst-sal.tuc.noao.edu/schema/SALEventSet.xsd">
+</SALEventSet>
+"""
+                )
```
Comment (r-owen) [inline python/lsst/ts/sal/make_salpy_lib.py:98]: Same comment as for make_idl_files.py

**Review by r-owen — APPROVED** [review]
Thank you for this update. I have a few changes I hope you will consider (and, in one case, try removing some code that I hope is no longer needed!) before merging.

File: bin/make_idl_files.py [hunk 5]
```diff
@@ -41,6 +41,14 @@
     components = lsst.ts.xml.subsystems
 else:
     components = args.components
+    invalid_components = [
+        component for component in components if component not in lsst.ts.xml.subsystems
+    ]
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply bin/make_idl_files.py:46]: That is nice! thanks for the suggestion!

File: bin/make_idl_files.py (hunk 1 above)
Comment (tribeiro) [reply bin/make_idl_files.py:47]: great catch! I forgot the `> 0`

File: python/lsst/ts/sal/make_idl_file.py (hunk 3 above)
Comment (tribeiro) [reply python/lsst/ts/sal/make_idl_file.py:81]: Unfortunately, it is needed. salgenerator fails if the file is not there.. :(
